src/Service/SF1500/SF1500.php

```diff
@@ -0,0 +1,859 @@
+            'ns3FiltreretOejebliksbillede',
+            'ns3Registrering',
+            'ns3RelationListe',
+            'ns2TilknyttedePersoner',
+            'ns2ReferenceID',
+            'ns2UUIDIdentifikator',
+        ];
+
+        $personId = $this->getValue($data, $personIdKeys, '');
+
+        if (null === $personId) {
+            return '';
+        }
+
+        $data = $this->personLaes($personId, $token);
+
+        $navnTekstKeys = [
+            'ns3LaesOutput',
+            'ns3FiltreretOejebliksbillede',
+            'ns3Registrering',
+            'ns3AttributListe',
+            'ns3Egenskab',
+            'ns3NavnTekst',
+        ];
+
+        return $this->getValue($data, $navnTekstKeys, '');
+    }
+
+    /**
+     * Fetches person phone from SF1500.
+     */
+    public function getPersonPhone(string $brugerId): string
+    {
+        return $this->getBrugerAdresseAttribut('Mobiltelefon_bruger', $brugerId);
+    }
+
+    /**
+     * Fetches person location from SF1500.
+     */
+    public function getPersonLocation(string $brugerId): string
+    {
+        return $this->getBrugerAdresseAttribut('Lokation_bruger', $brugerId);
+    }
+
+    /**
+     * Fetches person email from SF1500.
+     */
+    public function getPersonEmail(string $brugerId): string
+    {
+        return $this->getBrugerAdresseAttribut('Email_bruger', $brugerId);
+    }
+
+    /**
+     * Fetches organisations funktioner from SF1500.
+     */
+    public function getOrganisationFunktionerFromUserId(string $brugerId)
+    {
+        return $this->soegOrganisationFunktioner($brugerId, null, null, null);
+    }
+
+    /**
+     * Searches for organisations funktioner from SF1500.
+     */
+    public function soegOrganisationFunktioner(?string $brugerId, ?string $funktionsNavn, ?string $organisationId, ?string $funktionsTypeId)
+    {
+        $token = $this->fetchSAMLToken();
+
+        if (null === $token) {
+            return '';
+        }
+
+        $data = $this->organisationFunktionSoeg($brugerId, $funktionsNavn, $organisationId, $funktionsTypeId, $token);
+
+        $idListeKeys = [
+            'ns3SoegOutput',
+            'ns2IdListe',
+            'ns2UUIDIdentifikator',
+        ];
+
+        $id = $this->getValue($data, $idListeKeys);
+
+        return $id;
+    }
+
+    /**
+     * Fetches organisation enhed level 1 name from SF1500.
+     */
+    public function getOrganisationEnhed(string $funktionsId, bool $returnOrganisationID = false): ?string
+    {
+        $token = $this->fetchSAMLToken();
+
+        $data = $this->organisationFunktionLaes($funktionsId, $token);
+
+        $tilknyttedeEnhederKeys = [
+            'ns3LaesOutput',
+            'ns3FiltreretOejebliksbillede',
+            'ns3Registrering',
+            'ns3RelationListe',
+            'ns2TilknyttedeEnheder',
+            'ns2ReferenceID',
+            'ns2UUIDIdentifikator',
+        ];
+
+        $orgEnhedId = $this->getValue($data, $tilknyttedeEnhederKeys);
```

Comment:
Should we thro an exception if we cannot find a value for `$orgEnhedId`?

Reply:
Yes, fixed to throw exception.